src/AzureAuth.Test/CommandMainTest.cs

```diff
@@ -384,6 +389,142 @@ public void TestPromptHintPrefixWithoutOption()
+        public void TestCacheFileOptionWithInvalidFileCharacter()
+        {
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.CacheFilePath = "invalid char" + Path.GetInvalidPathChars().First();
+            subject.EvaluateOptions().Should().BeFalse();
+        }
+
+        /// <summary>
+        /// The test to evaluate a normal customized cache file name in enviroment variables.
+        /// </summary>
+        [Test]
+        public void TestCacheFileOptionWithNormalFilenameFromEnv()
+        {
+            string filenameFromEnv = "normal_file_name_from_env";
+            this.envMock.Setup(env => env.Get("AZUREAUTH_CACHE_FILE")).Returns(filenameFromEnv);
+
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.EvaluateOptions().Should().BeTrue();
+            subject.WrappedCacheFilePath.Should().Be(filenameFromEnv);
+        }
+
+        /// <summary>
+        /// The test to evaluate the cache file name when both enviroment variable and option exist.
+        /// Ideally use option first.
+        /// </summary>
+        [Test]
+        public void TestCacheFileOptionWithFilenameFromEnvAndOption()
+        {
+            string filenameFromEnv = "normal_file_name_from_env";
+            this.envMock.Setup(env => env.Get("AZUREAUTH_CACHE_FILE")).Returns(filenameFromEnv);
+
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.CacheFilePath = "normal_file_name_from_option";
+
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.EvaluateOptions().Should().BeTrue();
+            subject.WrappedCacheFilePath.Should().Be("normal_file_name_from_option");
+        }
+
+        /// <summary>
+        /// The test to evaluate a default cache file name.
+        /// </summary>
+        [Test]
+        public void TestCacheFileOptionWithNoParameter()
+        {
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.EvaluateOptions().Should().BeTrue();
+            subject.WrappedCacheFilePath.Should().Be($"msal_{subject.Tenant}.cache");

```

Comment:
This is where changing from WrappedCacheFilePath to just the CacheFilePath property should be asserted that it's the full absolute path. Given it will differ by user, the assertion will have to be a little bit more flexible but I think it can assert that it is a Rooted path, and ends with this filename.